packages/features/timezone-buddy/components/HoverOverview.tsx

```diff
@@ -0,0 +1,149 @@
+import { useState, useRef, useEffect, useMemo } from "react";
+
+import { DAY_CELL_WIDTH } from "../constants";
+
+function rounded(x: number, dayCellWidth: number) {
+  let n = Math.round(x / dayCellWidth);
+  n = Math.max(0, n);
+  n = Math.min(24, n);
+  return n * dayCellWidth;
+}
+function useElementBounding<T extends HTMLDivElement>(ref: React.RefObject<T>): DOMRect | null {
+  const [boundingRect, setBoundingRect] = useState<DOMRect | null>(null);
+  const observer = useRef<ResizeObserver | null>(null);
+
+  useEffect(() => {
+    const target = ref.current;
+
+    function updateBoundingRect() {
+      if (target) {
+        const rect = target.getBoundingClientRect();
+        setBoundingRect(rect);
+      }
+    }
+
+    updateBoundingRect();
+
+    observer.current = new ResizeObserver(updateBoundingRect);
+    observer.current.observe(target as Element);
+
+    return () => {
+      if (observer.current) {
+        observer.current.disconnect();
+        observer.current = null;
+      }
+    };
+  }, [ref]);
+
+  return boundingRect;
+}
+
+function useMouse() {
+  const [x, setX] = useState(0);
+  const [y, setY] = useState(0);
+  const [isPressed, setIsPressed] = useState(false);
+
+  useEffect(() => {
+    function handleMouseMove(event: MouseEvent) {
+      setX(event.clientX);
+      setY(event.clientY);
+    }
+
+    function handleMouseDown() {
+      setIsPressed(true);
+    }
+
+    function handleMouseUp() {
+      setIsPressed(false);
+    }
+
+    document.addEventListener("mousemove", handleMouseMove);
+    document.addEventListener("mousedown", handleMouseDown);
+    document.addEventListener("mouseup", handleMouseUp);
+
```

Comment:
should we use throttling for these events ?

Reply:
Great suggestion! I will deffo consider this when i come to implementing the hover view. Not 100% sure if its needed as the only thing this will be rendering is the x position of a box. So it won't be causing re-renders down the tree